### DISCUSSION TITLE
Wizard page type generation

### DISCUSSION THREAD

**Discussion by hamishwillee** [mdn/yari#4071] [Ideas]
I've been updating the page type templates linked from https://developer.mozilla.org/en-US/docs/MDN/Structures/Page_types - for example see https://github.com/mdn/content/pull/6217.

Wondering if it makes sense to add yari commands to auto-create pages using a wizard and (perhaps) jinja templates.

So workflow might be to do 
```
yarn add content template
```
- Then you get asked for template if you didn't specify one: api_ref, html_element, ...
- Then you get asked the thing's name - e.g. video
- From this the template would be able to generate the file because the locations are always the same. Also maybe a title, and definitely a stub. Might be able to work out bcd. 
- You could further ask "is this experimental, deprecated, etc etc."

Possibly a dumb idea, in particular as I understand some docs are auto-generated from IDL and other sources. But just feels like it might be a bit of a timesaver.

**Comment by peterbe** [comment]
One way of solving this would be a personal npm script instead of shipping it in Yari. 
E.g. `npx @hamish/page-type-wizard html_element ...` 
That script would be able to read from `CONTENT_ROOT` and use Yari APIs to find content and generate a new thing. 
Once that works, and it starts to prove itself popular and mature we can *then* add it to Yari's core. 

Not trying to shy away from the work, just saying it's a decent strategy to get momentum.

**Reply by Ryuno-Ki** [reply]
[`hygen`](https://www.npmjs.com/package/hygen) is popular in this area (as it combines the wizard with a template).

**Reply by hamishwillee** [reply]
Thanks - both seem good approaches. This was more a question in the first case of "would this be useful to you" and "would you prefer this to trying to copy-update the manual templates". My feeling is yes but perhaps not worth the development effort for the dev team vs all the other things they might be doing.
